### ISSUE TITLE
transforms to_pil_image clamp

### ISSUE THREAD

**Issue by irtez** [pytorch/vision#8235]
I was trying to convert images from tensor to save them on disk later and noticed strange behaviour in v2.ToPILImage.
If I save tensor calling save_image from torchvision.utils, it results in this:

![1216178672-1 (25)](https://github.com/pytorch/vision/assets/88113941/99b0d639-b974-4f3f-8475-8c212e09199b)

However, if I try to convert tensor by calling v2.ToPILImage and then save it with Image.save() thats what I get:

![1216178672-1](https://github.com/pytorch/vision/assets/88113941/3138c624-f1a3-43ca-bfff-b160d623c211)

As you can see, there are these strange bright pink areas.

Torchvision.utils saves the image this way:

https://github.com/pytorch/vision/blob/e0fd033c7db4f4e2e30d3452ed03ce35e097fc01/torchvision/utils.py#L149-L150

In v2.toPILImage there's following:

https://github.com/pytorch/vision/blob/e0fd033c7db4f4e2e30d3452ed03ce35e097fc01/torchvision/transforms/functional.py#L280-L281

so I manually did transform my image:
`ndarr = result_tensor.mul(255).add_(0.5).clamp_(0, 255).permute(1, 2, 0).to("cpu", torch.uint8).numpy()`
`result_img = Image.fromarray(ndarr)`

and the problem disappeared.

I don't know about adding 0.5 (I assume it is used for better rounding), but from what I understand clamp() is missing here.

**Comment by NicolasHug** [comment]
Thanks for the report @irtez 

What's the `min()` and `max()` values of the tensor image? It's possible that the float tensor isn't exactly within `[0, 1]` which might be causing the under/overflow issues

**Comment by irtez** [comment]
> Thanks for the report @irtez
> 
> What's the `min()` and `max()` values of the tensor image? It's possible that the float tensor isn't exactly within `[0, 1]` which might be causing the under/overflow issues

Yes, this is causing the problem, last layer of my net is convolutional and output tensors are not in [0, 1]. But shouldn't toPILImage clamp these out-of-range values like save_image to not cause problems with image?

**Comment by NicolasHug** [comment]
That's a fair point, but unfortunately, it's quite hard to please everyone here and for every user who would like the transfrom to clamp the input, we'd find another user who doesn't want the clamping to happen.

Always clamping and in every single transform is potentially a waste of time and negatively affects performance, for those users who know that the `[0, 1]` range is always respected on float tensors. So right now, we don't clamp of check manually, and rely on the (documented) assumption that the input is in `[0, 1]` if the image is a float. The drawback is that in some cases, it will lead to the issue you're experimenting though :/

Your best bet here would be to manually clamp to `[0, 1]` before calling `ToPilImage()` (or more generally, right after the output of your network)
